### PR TITLE
2021.1

### LINKED ISSUE #546: Update package pytest to 6.2
Update package pytest to 6.2

This release includes a very useful improvement to show the reason for a skip or xfail in verbose mode. This was a 4-year old issue that finally got fixed. In addition it has a new temp dir test fixture that it is a Path instead of a py.path.local thingy.

https://docs.pytest.org/en/stable/changelog.html#pytest-6-2-0-2020-12-12

On the flip side pytest makes so many breaking changes that we'll have to check that it doesn't break too many tests (though we don't really scratch the surface of pytest functionality).

### LINKED ISSUE #547: Update package bep_pcb_check to 3.2.0
Update package bep_pcb_check to 3.2.0 

Part of "New releases of ACIS thermal model check packages with updated test answers" for post-shiny release.

### LINKED ISSUE #548: Update package fep1_actel_check to 3.2.0
Update package fep1_actel_check to 3.2.0

Part of "New releases of ACIS thermal model check packages with updated test answers" for post-shiny release.

### LINKED ISSUE #549: Update package fep1_mong_check to 3.2.0
Update package fep1_mong_check to 3.2.0

Part of "New releases of ACIS thermal model check packages with updated test answers" for post-shiny release.

### LINKED ISSUE #550: Update package acisfp_check to 3.4.0
Update package acisfp_check to 3.4.0

Part of "New releases of ACIS thermal model check packages with updated test answers" for post-shiny release.

### LINKED ISSUE #551: Update package psmc_check to 3.2.0
Update package psmc_check to 3.2.0

Part of "New releases of ACIS thermal model check packages with updated test answers" for post-shiny release.

### LINKED ISSUE #552: Update package dea_check to 3.3.0
Update package dea_check to 3.3.0

Part of "New releases of ACIS thermal model check packages with updated test answers" for post-shiny release.

### LINKED ISSUE #553: Update package dpa_check to version 3.2.0
Update package dpa_check to version 3.2.0

Part of "New releases of ACIS thermal model check packages with updated test answers" for post-shiny release.

### LINKED ISSUE #556: Update sot/skare3_tools to 1.0.2
Includes the following merges:
- PR [#75](https://github.com/sot/skare3_tools/pull/75): added skare3_promote
- PR [#76](https://github.com/sot/skare3_tools/pull/76): skare3_tools.packages fixes for when ska3-matlab or ska3-flight are not in the conda channel
- PR [#74](https://github.com/sot/skare3_tools/pull/74): Improve info provided by ska3_update_summary
- PR [#73](https://github.com/sot/skare3_tools/pull/73): deal with default branch

### LINKED ISSUE #557: Update acisops/acis_thermal_check to 3.4.1
Includes the following merges:
- PR [#36](https://github.com/acisops/acis_thermal_check/pull/36): Make regression testing work on standalone platforms

### LINKED ISSUE #558: Release ska3-flight 2021.1
# Packages to include:

All [2021.1 milestone issues](https://github.com/sot/skare3/issues?q=is%3Aopen+is%3Aissue+milestone%3A2021.1)

# Checklist

- [x] Document all changes
- [x] Build all packages
   - [x] ~noarch~ (done by CI)
   - [x] ~linux 64~ (done by CI)
   - [x] ~mac 64~ (done by CI)
- [x] ~Copy Packages to test channel~ (done by CI)
- [x] Document test status
- [x] Get approval
- [x] Move individual packages to main conda channel
- [x] Update production systems:
   - [x] announce to aca@cfa
   - [x] move ska3-flight package to main conda channel
   - [x] update HEAD ska3/flight as aca user
   - [x] run ska_testr on HEAD
   - [x] ~update ska3/flight on cheru as sot user~
   - [x] update ska3/flight on chimchim as SOT user
   - [x] announce to chandracode and mpweekly

### LINKED ISSUE #559: Update sot/mica to Release 4.24.0
Includes the following merges:
- PR [#241](https://github.com/sot/mica/pull/241): Use vectorized Quat
- PR [#254](https://github.com/sot/mica/pull/254): A few small V&V and report updates
- PR [#252](https://github.com/sot/mica/pull/252): L0 ingest issues (both shiny and repro V)
- PR [#253](https://github.com/sot/mica/pull/253): Increase an ASOL quat vs OBC quat test tolerance by 4 arcsecs in roll
- PR [#251](https://github.com/sot/mica/pull/251): Add list of available content types to asp_l1.get_files
- PR [#250](https://github.com/sot/mica/pull/250): Minor V&V fixes
- PR [#249](https://github.com/sot/mica/pull/249): More centroid_dashboard tweaks

### LINKED ISSUE #560: Update sot/ska_helpers to Release 0.4.0
Includes the following merges:
- PR [#19](https://github.com/sot/ska_helpers/pull/19): add detail about all exceptions in ska_helpers.retry
- PR [#20](https://github.com/sot/ska_helpers/pull/20): Add lru_cache_timed decorator
- PR [#18](https://github.com/sot/ska_helpers/pull/18): Add retry and tests sub-packages
- PR [#17](https://github.com/sot/ska_helpers/pull/17): Add retry decorator and function wrapper

### LINKED ISSUE #561: Update sot/aca_weekly_report to Release 0.1.4
Includes the following merges:
- PR [#7](https://github.com/sot/aca_weekly_report/pull/7): Fix the type of the days-back option

### LINKED ISSUE #562: Update sot/eng_archive to Release 4.51.0
Includes the following merges:
- PR [#211](https://github.com/sot/eng_archive/pull/211): Move sync archive update into normal task schedule
- PR [#210](https://github.com/sot/eng_archive/pull/210): Update docs shiny
- PR [#209](https://github.com/sot/eng_archive/pull/209): Better caching in fetch

### LINKED ISSUE #563: Update sot/testr to Release 4.6.0
Includes the following merges:
- PR [#34](https://github.com/sot/testr/pull/34): Add functionality to skip tests via skip.yml in package dir

### LINKED ISSUE #564: Update sot/acdc to Release 4.7.1
Includes the following merges:
- PR [#59](https://github.com/sot/acdc/pull/59): Catch another source of Telemetry Gap Error

### LINKED ISSUE #565: Update sot/aca_view to Release 0.5.0
Includes the following merges:
- PR [#64](https://github.com/sot/aca_view/pull/64): Data service

### LINKED ISSUE #566: Update sot/maude to Release 3.5.0
Includes the following merges:
- PR [#31](https://github.com/sot/maude/pull/31): added blobs_to_table function
- PR [#29](https://github.com/sot/maude/pull/29): Allow access without credentials when possible
- PR [#30](https://github.com/sot/maude/pull/30): Use LazyDict from ska_helpers.utils

### LINKED ISSUE #567: Update sot/hopper to Release 4.5.1
Includes the following merges:
- PR [#20](https://github.com/sot/hopper/pull/20): Package/install the tests

### LINKED ISSUE #569: Update sot/aca_hi_bgd to Release 0.1.5
Includes the following merges:
- PR [#9](https://github.com/sot/aca_hi_bgd/pull/9): Rearrange processing at end to avoid reprocessing

### LINKED ISSUE #570: Update sot/kadi to Release 5.5.0
Includes the following merges:
- PR [#197](https://github.com/sot/kadi/pull/197): Add test of creating cmds archive and retry opening cmds.h5
- PR [#195](https://github.com/sot/kadi/pull/195): Add Ska.ParseCM compatibility + read_backstop function
- PR [#196](https://github.com/sot/kadi/pull/196): Handle cmds.h5 resource unavailable error

### LINKED ISSUE #571: Update sot/parse_cm to Release 3.6.0
Includes the following merges:
- PR [#30](https://github.com/sot/parse_cm/pull/30): Add cmds property to return kadi.commands.CommandTable version
- PR [#28](https://github.com/sot/parse_cm/pull/28): Support full parsing of obs request per OP19
- PR [#26](https://github.com/sot/parse_cm/pull/26): fixed bug skipping the last line of an OBS block
- PR [#29](https://github.com/sot/parse_cm/pull/29): Add docs and modernize code
- PR [#24](https://github.com/sot/parse_cm/pull/24): Shiny workflow update
- PR [#23](https://github.com/sot/parse_cm/pull/23): Shiny workflow update

### LINKED ISSUE #572: Update sot/proseco to Release 4.11.0
Includes the following merges:
- PR [#349](https://github.com/sot/proseco/pull/349): Built-in function for diffing ACA catalogs

### LINKED ISSUE #574: Update acisops/backstop_history to 1.4.0
Includes the following merges:
- PR [#18](https://github.com/acisops/backstop_history/pull/18): Replace Chandra.Time with cxotime

### LINKED ISSUE #575: Update acisops/acis_thermal_check to 3.5.0
Includes the following merges:
- PR [#37](https://github.com/acisops/acis_thermal_check/pull/37): Refactoring of limit handling, allowing for more customizations, use cxotime

### LINKED ISSUE #576: Update acisops/fep1_actel_check to 3.3.0
Includes the following merges:
- PR [#6](https://github.com/acisops/fep1_actel_check/pull/6): Update regression testing

### LINKED ISSUE #577: Update acisops/dea_check to 3.4.0
Includes the following merges:
- PR [#30](https://github.com/acisops/dea_check/pull/30): Update regression testing

### LINKED ISSUE #578: Update acisops/dpa_check to 3.3.0
Includes the following merges:
- PR [#28](https://github.com/acisops/dpa_check/pull/28): Add customizations to support checking +12 degC/0 FEPs violations, other minor updates

### LINKED ISSUE #579: Update acisops/fep1_mong_check to 3.3.0
Includes the following merges:
- PR [#6](https://github.com/acisops/fep1_mong_check/pull/6): Update regression testing

### LINKED ISSUE #580: Update acisops/bep_pcb_check to 3.3.0
Includes the following merges:
- PR [#6](https://github.com/acisops/bep_pcb_check/pull/6): Update regression testing

### LINKED ISSUE #581: Update acisops/psmc_check to 3.3.0
Includes the following merges:
- PR [#25](https://github.com/acisops/psmc_check/pull/25): Update regression testing

### LINKED ISSUE #582: Update acisops/acisfp_check to 3.5.0
Includes the following merges:
- PR [#29](https://github.com/acisops/acisfp_check/pull/29): Simplification of code, use cxotime, fixing bugs with violations reporting

### PR DESCRIPTION
**Pull request by javierggt**
# ska3-flight 2021.1

## Summary:

This is a routine update of the Ska3 runtime environment including a substantial number of package updates.

The ACIS load review tools will be briefed separately by ACIS.

## Interface Impacts:

No breaking changes, but these are some improvements of interest to Ska3 users that will be discussed. A detailed list of changes can be found in the 'code changes' section below.

- kadi 5.4.0
  - Add Ska.ParseCM compatibility + read_backstop function
- maude 3.4.0
  - Allow access without credentials when possible
- parse_cm 3.5.2
  - Add docs and modernize code
  - Support full parsing of obs request per OP19
  - Add cmds property to return kadi.commands.CommandTable version
- proseco 4.10.0
  - Built-in function for diffing ACA catalogs
- ska.engarchive 4.50.1
  - Better caching in fetch
- ska_helpers 0.3.0
  - Add retry decorator and function wrapper
  - Add lru_cache_timed decorator

## Testing:

This release is available for testing as ska3-flight version 2020.1rc3 from the test conda channel.

### Linux (HEAD)

Test results are displayed in [the dashboard](https://icxc.cfa.harvard.edu/aspect/skare3/testr/releases/2021.1rc3/)

Several tests are noted as failing in the dashboard:

- acisfp_check-tests and psmc_check post_check_logs.log fails: “ErfaWarning: ERFA function “dtf2d” yielded 2 of “dubious year (Note 6)” and similar.

  This is a benign warning related to the astropy release being over a year old and being worried that a leap second might have occurred. It has not, so OK. We plan to update to a more recent astropy soon.
- dpa_check post_check_logs.log fails: 1dpamzt violates zero-feps limit of 12.00 degC…

  This is just our testing tool not expecting this new warning. We will fix.
  
- mica test fails because the runner does not have sybase access.

  We are working on sybase access for the test runner account. By-hand run is OK.

### MacOSX standalone

All tests pass with exception of the same ERFA warnings.

### Windows 10 standalone

All tests pass with one exception.

The `hopper` package fails tests because it has embedded strings in the unit tests that are separated by `\n` but the package is expected Windows line separators `os.linesep`. This is acceptable because the `hopper` package is currently only used by `starcheck`, which does not run on Windows. This testing is newly added in this release of Ska3.

## Review:

All operations critical or impacting PR's are independently and carefully reviewed. For other PR's the level of detail for review is calibrated to operations criticality. Some PR's that are confined to aspect-team-specific processing may have little to no independent review.

## Deployment:
   *  The 2020.10 ska3-flight release will be deployed on HEAD and GRETA ska3/flight after the release and products for the week are approved.
   
# Code changes

### Updated Packages

- **pytest:** 5.4.3 -> 6.2.0

### Updated Packages

- **aca_hi_bgd:** 0.1.4 -> 0.1.5 (0.1.4 -> 0.1.5)
  - [PR 9](https://github.com/sot/aca_hi_bgd/pull/9) (jeanconn): Rearrange processing at end to avoid reprocessing
- **aca_view:** 0.4.1 -> 0.5.0 (0.4.1 -> 0.5.0)
  - [PR 64](https://github.com/sot/aca_view/pull/64) (javierggt): Improved data fetching
- **aca_weekly_report:** 0.1.3 -> 0.1.4 (0.1.3 -> 0.1.4)
  - [PR 7](https://github.com/sot/aca_weekly_report/pull/7) (jeanconn): Fix the type of the days-back option
- **acdc:** 4.7.0 -> 4.7.1 (4.7.0 -> 4.7.1)
  - [PR 59](https://github.com/sot/acdc/pull/59) (taldcroft): Catch another source of Telemetry Gap Error
- **acis_thermal_check:** 3.4.0 -> 3.5.0 (3.4.0 -> 3.4.1 -> 3.5.0)
  - [PR 36](https://github.com/acisops/acis_thermal_check/pull/36) (jzuhone): Make regression testing work on standalone platforms
  - [PR 37](https://github.com/acisops/acis_thermal_check/pull/37) (jzuhone): Refactoring of limit handling, allowing for more customizations, use cxotime
- **acisfp_check:** 3.2.0 -> 3.5.0 (3.2.0 -> 3.3.0 -> 3.4.0 -> 3.5.0)
  - [PR 27](https://github.com/acisops/acisfp_check/pull/27) (jzuhone): Refactor tests to match new signature
  - [PR 28](https://github.com/acisops/acisfp_check/pull/28) (jzuhone): Update answers post-shiny
  - [PR 29](https://github.com/acisops/acisfp_check/pull/29) (jzuhone): Simplification of code, use cxotime, fixing bugs with violations reporting
- **backstop_history:** 1.3.0 -> 1.4.0 (1.3.0 -> 1.4.0)
  - [PR 18](https://github.com/acisops/backstop_history/pull/18) (jzuhone): Replace Chandra.Time with cxotime
- **bep_pcb_check:** 3.0.0 -> 3.3.0 (3.0.0 -> 3.1.0 -> 3.2.0 -> 3.3.0)
  - [PR 4](https://github.com/acisops/bep_pcb_check/pull/4) (jzuhone): Refactor tests to match new signature
  - [PR 5](https://github.com/acisops/bep_pcb_check/pull/5) (jzuhone): Update answers for shiny
  - [PR 6](https://github.com/acisops/bep_pcb_check/pull/6) (jzuhone): Update regression testing
- **dea_check:** 3.1.0 -> 3.4.0 (3.1.0 -> 3.2.0 -> 3.3.0 -> 3.4.0)
  - [PR 28](https://github.com/acisops/dea_check/pull/28) (jzuhone): Refactor tests to match new signature
  - [PR 29](https://github.com/acisops/dea_check/pull/29) (jzuhone): Update answers for shiny
  - [PR 30](https://github.com/acisops/dea_check/pull/30) (jzuhone): Update regression testing
- **dpa_check:** 3.0.0 -> 3.3.0 (3.0.0 -> 3.1.0 -> 3.2.0 -> 3.3.0)
  - [PR 26](https://github.com/acisops/dpa_check/pull/26) (jzuhone): Refactor tests to match new signature
  - [PR 27](https://github.com/acisops/dpa_check/pull/27) (jzuhone): Update answers for shiny
  - [PR 28](https://github.com/acisops/dpa_check/pull/28) (jzuhone): Add customizations to support checking +12 degC/0 FEPs violations, other minor updates
- **fep1_actel_check:** 3.0.0 -> 3.3.0 (3.0.0 -> 3.1.0 -> 3.2.0 -> 3.3.0)
  - [PR 4](https://github.com/acisops/fep1_actel_check/pull/4) (jzuhone): Refactor tests to match new signature
  - [PR 5](https://github.com/acisops/fep1_actel_check/pull/5) (jzuhone): Update answers for shiny
  - [PR 6](https://github.com/acisops/fep1_actel_check/pull/6) (jzuhone): Update regression testing
- **fep1_mong_check:** 3.0.0 -> 3.3.0 (3.0.0 -> 3.1.0 -> 3.2.0 -> 3.3.0)
  - [PR 4](https://github.com/acisops/fep1_mong_check/pull/4) (jzuhone): Refactor tests to match new signature
  - [PR 5](https://github.com/acisops/fep1_mong_check/pull/5) (jzuhone): Update answers for shiny
  - [PR 6](https://github.com/acisops/fep1_mong_check/pull/6) (jzuhone): Update regression testing
- **hopper:** 4.5.0 -> 4.5.1 (4.5.0 -> 4.5.1)
  - [PR 20](https://github.com/sot/hopper/pull/20) (jeanconn): Package/install the tests
- **kadi:** 5.4.0 -> 5.5.0 (5.4.0 -> 5.5.0)
  - [PR 196](https://github.com/sot/kadi/pull/196) (taldcroft): Handle cmds.h5 resource unavailable error
  - [PR 195](https://github.com/sot/kadi/pull/195) (taldcroft): ***Add Ska.ParseCM compatibility + read_backstop function***
  - [PR 197](https://github.com/sot/kadi/pull/197) (taldcroft): Add test of creating cmds archive and retry opening cmds.h5
- **maude:** 3.4.0 -> 3.5.0 (3.4.0 -> 3.5.0)
  - [PR 30](https://github.com/sot/maude/pull/30) (taldcroft): Use LazyDict from ska_helpers.utils
  - [PR 29](https://github.com/sot/maude/pull/29) (taldcroft): ***Allow access without credentials when possible***
  - [PR 31](https://github.com/sot/maude/pull/31) (javierggt): added blobs_to_table function
- **mica:** 4.23.0 -> 4.24.0 (4.23.0 -> 4.24.0)
  - [PR 249](https://github.com/sot/mica/pull/249) (jeanconn): More centroid_dashboard tweaks
  - [PR 250](https://github.com/sot/mica/pull/250) (jeanconn): Minor V&V fixes
  - [PR 251](https://github.com/sot/mica/pull/251) (taldcroft): Add list of available content types to asp_l1.get_files
  - [PR 253](https://github.com/sot/mica/pull/253) (jeanconn): Increase an ASOL quat vs OBC quat test tolerance by 4 arcsecs in roll
  - [PR 252](https://github.com/sot/mica/pull/252) (jeanconn): L0 ingest issues (both shiny and repro V)
  - [PR 254](https://github.com/sot/mica/pull/254) (taldcroft): A few small V&V and report updates
  - [PR 241](https://github.com/sot/mica/pull/241) (taldcroft): Use vectorized Quat
- **parse_cm:** 3.5.2 -> 3.6.0 (3.5.2 -> 3.6.0)
  - [PR 29](https://github.com/sot/parse_cm/pull/29) (taldcroft): ***Add docs and modernize code***
  - [PR 26](https://github.com/sot/parse_cm/pull/26) (taldcroft): fixed bug skipping the last line of an OBS block
  - [PR 28](https://github.com/sot/parse_cm/pull/28) (taldcroft): ***Support full parsing of obs request per OP19***
  - [PR 30](https://github.com/sot/parse_cm/pull/30) (taldcroft): ***Add cmds property to return kadi.commands.CommandTable version***
- **proseco:** 4.10.0 -> 4.11.0 (4.10.0 -> 4.11.0)
  - [PR 349](https://github.com/sot/proseco/pull/349) (taldcroft): ***Built-in function for diffing ACA catalogs***
- **psmc_check:** 3.0.0 -> 3.3.0 (3.0.0 -> 3.1.0 -> 3.2.0 -> 3.3.0)
  - [PR 23](https://github.com/acisops/psmc_check/pull/23) (jzuhone): Refactor tests to match new signature
  - [PR 24](https://github.com/acisops/psmc_check/pull/24) (jzuhone): Updating answers for shiny
  - [PR 25](https://github.com/acisops/psmc_check/pull/25) (jzuhone): Update regression testing
- **ska.engarchive:** 4.50.1 -> 4.51.0 (4.50.1 -> 4.51.0)
  - [PR 209](https://github.com/sot/eng_archive/pull/209) (taldcroft): ***Better caching in fetch***
  - [PR 210](https://github.com/sot/eng_archive/pull/210) (taldcroft): Update docs shiny
  - [PR 211](https://github.com/sot/eng_archive/pull/211) (taldcroft): Move sync archive update into normal task schedule
- **ska_helpers:** 0.3.0 -> 0.4.0 (0.3.0 -> 0.4.0)
  - [PR 17](https://github.com/sot/ska_helpers/pull/17) (taldcroft): ***Add retry decorator and function wrapper***
  - [PR 18](https://github.com/sot/ska_helpers/pull/18) (taldcroft): Add retry and tests sub-packages
  - [PR 20](https://github.com/sot/ska_helpers/pull/20) (taldcroft): ***Add lru_cache_timed decorator***
  - [PR 19](https://github.com/sot/ska_helpers/pull/19) (javierggt): add detail about all exceptions in ska_helpers.retry
- **skare3_tools:** 1.0.1 -> 1.0.2 (1.0.1 -> 1.0.2)
  - [PR 73](https://github.com/sot/skare3_tools/pull/73) (javierggt): deal with default branch name different than 'master'
  - [PR 74](https://github.com/sot/skare3_tools/pull/74) (javierggt): Improve info provided by ska3_update_summary
  - [PR 76](https://github.com/sot/skare3_tools/pull/76) (javierggt): skare3_tools.packages fixes for when ska3-matlab or ska3-flight are not in the conda channel
  - [PR 75](https://github.com/sot/skare3_tools/pull/75) (javierggt): added skare3-promote script
- **testr:** 4.5.0 -> 4.6.0 (4.5.0 -> 4.6.0)
  - [PR 34](https://github.com/sot/testr/pull/34) (taldcroft): Add functionality to skip tests via skip.yml in package dir

# Related Issues

Fixes #546
Fixes #547
Fixes #548
Fixes #549
Fixes #550
Fixes #551
Fixes #552
Fixes #553
Fixes #556
Fixes #557
Fixes #558
Fixes #559
Fixes #560
Fixes #561
Fixes #562
Fixes #563
Fixes #564
Fixes #565
Fixes #566
Fixes #567
Fixes #569
Fixes #570
Fixes #571
Fixes #572
Fixes #574
Fixes #575
Fixes #576
Fixes #577
Fixes #578
Fixes #579
Fixes #580
Fixes #581
Fixes #582

### CODE REVIEW

File: pkg_defs/ska3-core/meta.yaml [hunk 1]
```diff
@@ -197,7 +197,7 @@ requirements:
     - pyrsistent ==0.16.0
     - pysocks ==1.7.1
     - pytables ==3.6.1
-    - pytest ==5.4.3
+    - pytest ==6.2
```
Comment (taldcroft) [inline pkg_defs/ska3-core/meta.yaml:200]: The ska3-flight meta still references 2020.14 for core. We definitely want to check that all our tests are still working with 6.2.

File: pkg_defs/ska3-flight/meta.yaml [hunk 2]
```diff
@@ -1,50 +1,50 @@
 ---
 package:
   name: ska3-flight
-  version: 2020.14
+  version: 2021.1
```
Comment (jeanconn) [inline pkg_defs/ska3-flight/meta.yaml:4]: Should we also just add a comment to the meta.yaml with "# For nominal releases, update ska3-core in the package list to match ska3-flight version"?  I think this is the one extra manual step that *could* be done in code but we'll probably do by hand every time.

File: pkg_defs/ska3-core/meta.yaml (hunk 1 above)
Comment (javierggt) [reply pkg_defs/ska3-core/meta.yaml:200]: Fixed

File: pkg_defs/ska3-flight/meta.yaml (hunk 2 above)
Comment (javierggt) [reply pkg_defs/ska3-flight/meta.yaml:4]: yes, I added a comment to that effect on ska3-core meta.yaml.

**Pull request merged**